### PR TITLE
Known issue guests app

### PR DESCRIPTION
**Pull request by enbrnz**
https://github.com/owncloud/enterprise/issues/5428

Not sure if my proposed list is complete, please double check

Backport to 10.11 and 10.10

### CODE REVIEW

**Comment by mmattel** [comment]
@enbrnz I have improved the text a bit, but one question remains: what happens if the admin does not maintain the applist upfront and upgrades the guest app?

**Comment by phil-davis** [comment]
> @enbrnz I have improved the text a bit, but one question remains: what happens if the admin does not maintain the applist upfront and upgrades the guest app?

If the installation is using apps that "are inserted in the backend path to files" (such as encryption, S3 storage, probably ransomware protection, antivirus...) then guest users are broken - the app(s) that need to work on their behalf to serve files to them are not enabled for them to use.

https://github.com/owncloud/guests/pull/527 proposes to add some apps to the hard-coded minimal white-list to avoid this problem ever happening.

**Comment by mmattel** [comment]
Proposal: lets wait until Guests 0.12.1 is out and rework the changes based on the new behaviour.

**Comment by enbrnz** [comment]
> Proposal: lets wait until Guests 0.12.1 is out and rework the changes based on the new behaviour.

Is this going to happen today? There are customers having issues and creating tickets because this known issue is not properly communicated.

**Comment by jnweiger** [comment]
Yes, I just uploaded 0.12.1 RC1 to https://github.com/owncloud/guests/releases
give me a few minutes to make it not explode.

**Comment by pako81** [comment]
Is this doc issue still needed? All the mentioned apps are now hard-coded into the guests whitelists.

**Comment by jnweiger** [comment]
With 0.12.1 all these apps are in the CORE_WHITELIST. They are always whitelisted for guests. Admin cannot disable that.

* admin_audit 
* encryption
* files_antivirus
* firewall
* ransomware_protection

No need any more too mention a manual workaround for these in the docs. Please adjust this PR.

With guests 0.12.1, the complete whitelists are:
```
CORE_WHITELIST = ',core,files,guests,encryption,files_primary_s3,files_antivirus,files_external,files_external_dropbox,files_external_ftp,files_ldap_home,files_onedrive,sharepoint,files_external_s3,windows_network_drive,admin_audit,firewall,ransomware_protection';
DEFAULT_WHITELIST = 'settings,avatar,files_trashbin,files_versions,files_sharing,files_texteditor,activity,firstrunwizard,gallery,notifications,password_policy,oauth2,files_pdfviewer,files_mediaviewer,richdocuments,onlyoffice,wopi,oco_selfservice,twofactor_totp';
```

**Comment by mmattel** [comment]
Shall we add the whitelists as listed above to the documentation?
(Q: what is the difference between core and default whitelist?)

**Comment by enbrnz** [comment]
Is the complete tarball going to receive an update?
If not, I think we should say that the release still has a known issue and the app just needs an update.
If yes, this PR is redundant.

File: modules/ROOT/pages/server_release_notes.adoc [hunk 1]
```diff
@@ -100,6 +100,21 @@ Currently known incompatible apps include:
...
+* encryption
+* files_antivirus
+* firewall
+* ransomware_protection
+
+Any apps that provide access to external storage the guest users should have access too, need to be added as well like:
+
+* files_primary_s3
+* windows_network_drive
+
```
Comment (mmattel) [inline modules/ROOT/pages/server_release_notes.adoc:117]:
```suggestion
==== Guests App Whitelist Feature

The _Guests App Whitelist feature_ evaluates the app whitelist stricter starting with release 0.12.0 . To overcome any issues related to this strict handling, an upgrade to Guests 0.12.1 is highly recommended as this release has a much wider range of default whitelisted apps included.

```

Comment (mmattel) [reply modules/ROOT/pages/server_release_notes.adoc:117]: The tarball does not get an upgrade, we need to add this to the known issues section.

@EParzefall @phil-davis mind for a language review?

**Review by EParzefall — APPROVED** [review]

**Review by mmattel — APPROVED** [review]

**Pull request merged**